projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -12,12 +12,56 @@
 from typing import List
 
 from taurus_datajob_api import ApiException
+from vdk.internal.core import errors
 from vdk.plugin.meta_jobs.cached_data_job_executor import TrackingDataJobExecutor
 from vdk.plugin.meta_jobs.meta import TrackableJob
 from vdk.plugin.meta_jobs.remote_data_job_executor import RemoteDataJobExecutor
 from vdk.plugin.meta_jobs.time_based_queue import TimeBasedQueue
 
 log = logging.getLogger(__name__)
+max_starting_jobs = 15
+
+
+def validate_job_limit(jobs: List[Dict]):
+    topo_sorter = TopologicalSorter()
+    for job in jobs:
+        topo_sorter.add(job["job_name"], *job["depends_on"])
+
+    sorted_jobs = []
+    for job_name in topo_sorter.static_order():
+        job = next((j for j in jobs if j["job_name"] == job_name), None)
+        if job:
+            sorted_jobs.append(job)
+
+    out_degree = {job["job_name"]: 0 for job in jobs}
+    for job in sorted_jobs:
+        for pred in job["depends_on"]:
+            print(pred)
+            out_degree[pred] += 1
+
+    ready_jobs = set()
+    for job in sorted_jobs:
+        if out_degree[job["job_name"]] == 0:
+            ready_jobs.add(job["job_name"])
+
+    for job in sorted_jobs:
```

Comment:
nit: a lot of loops, it's a good code style to comment on each code block (why necessary, what is the purpose), to make it easily readable.
then, you can think if optimisations (doing same stuff by reusing same loop) are possible